feat: zeko providers

mich3lang3lo opened the pull request:
## Describe changes
Starting the Zeko integration. Changing provider naming to check readiness of codebase to allow switching from Mina L1 to Zeko L2. 

Changes include new set of tests for Zeko with account information and also submission of transaction. Also adding to this a new package in `providers` called `mina-node` because existing one is fixed to `mina-explorer`.

🚀 🦋

mrcnk commented on packages/providers/src/mina-node/utils/json-serializer.ts:
```diff
@@ -0,0 +1,3 @@
+import JSONbig from 'json-bigint'
```
For JSON serialization with extra types we use `superjson`. It appears in other packages.

mich3lang3lo replied:
This is just a copy of the old `mina-explorer` sub-package, I will work on switching to `superjson` in another PR. Starting small.

mrcnk commented on packages/providers/src/mina-node/utils/custom-fetch.ts:
```diff
@@ -0,0 +1,18 @@
+import JSONbig from 'json-bigint'
```
Replace with `superjson`.

mrcnk commented on packages/providers/src/mina-node/utils/custom-fetch.ts:
```diff
@@ -0,0 +1,18 @@
+import JSONbig from 'json-bigint'
+
+export const customFetch = async (
```
Is adding a custom fetcher for `graphql-request` actually needed? Sorry, I don't get the purpose.

mich3lang3lo replied:
Yeah its just a copy of what was here before, I can improve this later maybe.

mrcnk commented on packages/providers/src/mina-node/utils/custom-fetch.ts:
```diff
@@ -0,0 +1,18 @@
+import JSONbig from 'json-bigint'
+
+export const customFetch = async (
```
Is adding a custom fetcher for `graphql-request` actually needed? Sorry, I don't get the purpose.

mrcnk reviewed (changes requested):
Solid work, keep it up Mariusz, and I'm glad to see a new contributor 🎉 left few comments

mich3lang3lo commented:
Hi @mrcnk I have fixed the failing test. It looks like there was a restart of Zeko since I made the PR. I fixed this by sending test Mina to the known Pallad account. Thanks for your support <3
